commercetools-models/src/test/java/io/sphere/sdk/categories/commands/CategoryDeleteCommandIntegrationTest.java

```diff
@@ -30,4 +31,26 @@ public void referenceExpansion() throws Exception {
         assertThat(deletedCategory.getParent().getObj()).isNotNull().isEqualTo(parent);
         client().executeBlocking(CategoryDeleteCommand.of(parent));
     }
+
+    @Test
+    public void executionForDeletionByKey() throws Exception {
+        final String key = randomKey();
+        final CategoryDraft draft = CategoryDraftBuilder.of(randomSlug(), randomSlug()).key(key).build();
+        final Category category = client().executeBlocking(CategoryCreateCommand.of(draft));
+
+        client().executeBlocking(CategoryDeleteCommand.ofKey(key,category.getVersion()));
+        assertThat(client().executeBlocking(CategoryQuery.of().byId(category.getId())).head()).isEmpty();
+    }
+
+    @Test
+    public void referenceExpansionForDeletionByKey() throws Exception {
+        final String key = randomKey();
+        final Category parent = client().executeBlocking(CategoryCreateCommand.of(CategoryDraftBuilder.of(randomSlug(), randomSlug()).build()));
+        final Category category = client().executeBlocking(CategoryCreateCommand.of(CategoryDraftBuilder.of(randomSlug(), randomSlug()).key(key).parent(parent).build()));
```

Comment:
How about replacing the first `randomSlug()` call to a new method `SphereTestUtils.randomLocalizedString()`?